giskard/scanner/issues.py

```diff
@@ -25,6 +26,42 @@ class IssueGroup:
     description: str
 
 
+class ScanExamples(ABC):
+    """
+    Abstract class to manage examples from different data types.
+    """
+
+    @abstractmethod
+    def add_examples(self, example: Any):
+        ...
+
+    @abstractmethod
+    def head(self, n: int):
+        ...
+
+    @abstractmethod
+    def to_html(self):
+        ...
+
+
+class DataFrameScanExamples(ScanExamples):
+    """
+    ScanExamples class to manage examples from pandas dataframes
+    """
+
+    def __init__(self, examples: pd.DataFrame = None):
+        self.examples = pd.DataFrame() if examples is None else examples
```

Comment:
Should we check if examples is a dataframe ?

Reply:
sure, we can add the `@configured_validate_arguments` on the `__init__`